### PR TITLE
Fix the `tbds-strip-unit` function

### PR DESCRIPTION
**Pull request by smaboshe**
CI is currently breaking because of the way that `tbds-strip-unit` has been implemented.

This commit updates the implementation of the `tbds-strip-unit` function.

Ref:
- https://app.circleci.com/pipelines/github/thoughtbot/design-system/121/workflows/bf168ccf-8c89-455a-8c42-e956acbf1ce3/jobs/456/parallel-runs/0/steps/0-105
- https://github.com/sass-collective/sass-collective/blob/master/packages/strip-unit/_functions.scss

### CODE REVIEW

**Comment by smaboshe** [comment]
> Can we confirm this doesn't break other things in other repos?

Good call, @rtaibah! Confirmed. We have not created a new release yet so other projects are not affected.

**Comment by smaboshe** [comment]
My understanding is that this functions implementation is inspired by this article:
https://css-tricks.com/snippets/sass/strip-unit-function/

File: src/settings/tools/strip-unit.scss [hunk 1]
```diff
@@ -1,3 +1,12 @@
+// Ref: https://github.com/sass-collective/sass-collective/blob/master/packages/strip-unit/_functions.scss
+@use "sass:math";
+@use "sass:meta";
+
 @function tbds-strip-unit($value) {
-  @return (calc($value / ($value * 0 + 1)));
+  @if meta.type-of($value) != "number" {
+    @error "Invalid `#{meta.type-of($value)}` type. Choose a number type instead.";
+  } @else if meta.type-of($value) == "number" and not math.is-unitless($value) {
```
Comment (LkeMitchll) [inline src/settings/tools/strip-unit.scss:8]: If the number passed is already unitless should we still return something? Or maybe throw an error/warning (my SASS isn't good enough to know if it has a warning concept)? 

If I were using this function I'd probably like to be told that I don't need to be using it 😆

File: src/settings/tools/strip-unit.scss [hunk 2]
```diff
@@ -1,3 +1,12 @@
+// Ref: https://github.com/sass-collective/sass-collective/blob/master/packages/strip-unit/_functions.scss
```
Comment (LkeMitchll) [inline src/settings/tools/strip-unit.scss:1]: Could we use this dependency and use it directly instead?

Comment (smaboshe) [reply src/settings/tools/strip-unit.scss:1]: I like this idea. I'll try to set this up and will reach out for help f I get stuck 🙂.

**Comment by smaboshe** [comment]
Closing this in favour of https://github.com/thoughtbot/design-system/pull/253

**Pull request closed without merging**